### PR TITLE
Shell gzip

### PR DESCRIPTION
**Pull request by aaron-prindle**

### CODE REVIEW

**Comment by dlorenc** [comment]
Have you had a chance to benchmark this before and after the change?

File: ftl/common/builder.py [hunk 1]
```diff
@@ -75,21 +75,19 @@ def CreatePackageBase(self, base_image, cache):
...
 
         tar = buf.getvalue()
         sha = 'sha256:' + hashlib.sha256(tar).hexdigest()
 
-        gz = cStringIO.StringIO()
-        logging.info('Starting to gzip tarfile...')
-        with gzip.GzipFile(fileobj=gz, mode='w', compresslevel=1) as f:
-            f.write(tar)
+        logging.info('Starting to gzip app layer tarfile...')
+        subprocess.check_call(['gzip', tar])
```
Comment (dlorenc) [inline ftl/common/builder.py]: 'tar' is just a string at this point. You'll have to pass it into stdin of subprocess via subprocess.communicate:
https://stackoverflow.com/questions/8475290/how-do-i-write-to-a-python-subprocess-stdin

**Comment by aaron-prindle** [comment]
Looks like a small performance improvement from the benchmarks:
```
shell-gzip
1 | 2017-11-19 19:09:20 UTC | large_app | 34.6 |  
-- | -- | -- | -- | --
2 | 2017-11-19 19:08:44 UTC | medium_app | 10.6 |  
3 | 2017-11-19 19:08:33 UTC | small_app | 7.37 |  

master
4 | 2017-11-19 18:55:31 UTC | large_app | 38.64 |  
5 | 2017-11-19 18:54:52 UTC | medium_app | 10.95 |  
6 | 2017-11-19 18:54:40 UTC | small_app | 7.76
```

File: ftl/common/builder.py [hunk 2]
```diff
@@ -65,21 +65,23 @@ def CreatePackageBase(self, base_image, cache):
...
 
         tar = buf.getvalue()
         sha = 'sha256:' + hashlib.sha256(tar).hexdigest()
 
-        gz = cStringIO.StringIO()
-        logging.info('Starting to gzip tarfile...')
-        with gzip.GzipFile(fileobj=gz, mode='w', compresslevel=1) as f:
-            f.write(tar)
+        logging.info('Starting to gzip app layer tarfile...')
+        gzip_process = subprocess.Popen(['gzip', '-f'],
```
Comment (dlorenc) [inline ftl/common/builder.py]: Nice, we have this in a few places now so it probably makes sense to refactor, but that can come in a different PR.

**Pull request merged**